rs/types/management_canister_types/src/lib.rs

```diff
@@ -808,6 +835,174 @@ impl From<pb_canister_state_bits::LogVisibility> for LogVisibility {
+        match item {
+            LogVisibilityV2::Controllers => pb::LogVisibilityV2 {
+                log_visibility_v2: Some(pb::log_visibility_v2::LogVisibilityV2::Controllers(1)),
+            },
+            LogVisibilityV2::Public => pb::LogVisibilityV2 {
+                log_visibility_v2: Some(pb::log_visibility_v2::LogVisibilityV2::Public(2)),
+            },
+            LogVisibilityV2::AllowedViewers(principals) => pb::LogVisibilityV2 {
+                log_visibility_v2: Some(pb::log_visibility_v2::LogVisibilityV2::AllowedViewers(
+                    pb::LogVisibilityAllowedViewers {
+                        principals: principals
+                            .get()
+                            .iter()
+                            .map(|c| (*c).into())
+                            .collect::<Vec<ic_protobuf::types::v1::PrincipalId>>()
+                            .clone(),
+                    },
+                )),
+            },
+        }
+    }
+}
+
+impl TryFrom<pb_canister_state_bits::LogVisibilityV2> for LogVisibilityV2 {
+    type Error = ProxyDecodeError;
+
+    fn try_from(item: pb_canister_state_bits::LogVisibilityV2) -> Result<Self, Self::Error> {
+        use pb_canister_state_bits as pb;
+        let Some(log_visibility_v2) = item.log_visibility_v2 else {
+            return Err(ProxyDecodeError::MissingField(
+                "LogVisibilityV2::log_visibility_v2",
+            ));
+        };
+        match log_visibility_v2 {
+            pb::log_visibility_v2::LogVisibilityV2::Controllers(_) => Ok(Self::Controllers),
+            pb::log_visibility_v2::LogVisibilityV2::Public(_) => Ok(Self::Public),
+            pb::log_visibility_v2::LogVisibilityV2::AllowedViewers(data) => {
+                let data = data
+                    .principals
+                    .iter()
+                    .map(|p| {
+                        PrincipalId::try_from(p.raw.clone()).map_err(|e| {
+                            ProxyDecodeError::ValueOutOfRange {
+                                typ: "PrincipalId",
+                                err: e.to_string(),
+                            }
+                        })
+                    })
+                    .collect::<Result<Vec<PrincipalId>, _>>()?;
+                Ok(Self::AllowedViewers(BoundedAllowedViewers::new(data)))
+            }
+        }
+    }
+}
+
+#[test]
+fn test_log_visibility_v2_serialization_roundtrip() {
+    // Test `LogVisibilityV2` canidid serialization/deserialization.
+    for (initial, expected) in [
+        (LogVisibilityV2::Controllers, LogVisibilityV2::Controllers),
+        (LogVisibilityV2::Public, LogVisibilityV2::Public),
+        (
+            LogVisibilityV2::AllowedViewers(Default::default()),
+            LogVisibilityV2::default(),
+        ),
+    ] {
+        let encoded = LogVisibilityV2::encode(&initial);
+        let decoded = LogVisibilityV2::decode(&encoded).unwrap();
+        assert_eq!(decoded, expected);
+    }
+}
```

Comment:
I don't think we need this as we don't implement any logic by hand for Candid (unlike protobuf where we have `From/Try_From` implementations that should be tested).

Reply:
This test is to make sure that `AllowedViewers` cannot get into replica from a user, but it is switched to default `Controllers` and is always compatible with existing `LogVisibility` v1.

Reply:
Ack, so we'll remove it then once we start allowing setting `AllowedViewers`?